eth/handler.go

```diff
@@ -250,8 +250,17 @@ func (h *handler) runEthPeer(peer *eth.Peer, handler eth.Handler) error {
 		peer.Log().Debug("Ethereum handshake failed", "err", err)
 		return err
 	}
+	reserved := 0 // reserved peer slots
+	if atomic.LoadUint32(&h.snapSync) == 1 && !peer.SupportsCap("snap", 1) {
+		// If we are running snap-sync, we want to reserve half the peer slots for
+		// peers supporting the snap protocol
+		reserved = h.maxPeers/2 - len(h.peers.snapPeers)
```

Comment:
`peers` needs a lock before you can access an internal field. We already have a `Len` for the `eth` peers, we can add a second one for the snap peers.